### ISSUE TITLE
After updating OpenELEC 6.0.0 to 6.0.1 KODI hangs on boot

### ISSUE THREAD

**Issue by dlabz** [OpenELEC/OpenELEC.tv#4668]
I was running OpenELEC 6.0.0 [generic x86_64] on Zotac ZBOX AD10 with 4GB RAM. Due known issue with Atheros WiFi adapter, I switched in the Broadcom BCM943225HM. Everything was working as expected.

After updating to OpenELEC 6.0.1, the system would hang once KODI starts. The system would be inaccessible via SSH. I also tried booting OpenELEC 6.0.1 from live usb drive, and it would work until I connect to Wifi.
That lead me to test booting my system with WiFI access point unplugged, and system would boot as expected.  
Than I removed the Broadcom BCM943225HM and put in the original Atheros adapter and OpenELEC 6.0.1 works as expected. 

TLDR:
OpenELEC 6.0.1 would hang when Broadcom BCM943225HM is present.

**Comment by chewitt** [comment]
which broadcom driver is this device attempting to use .. brcmfmac or wl?

**Comment by dlabz** [comment]
I have no idea how to get that information.
I can't access the machine via network when the BCM card is in slot.
What I did is to blacklist the machine on the router, and KODI loads, than use kodi file manager to manually copy the log to a thumb drive. But....
Logs folder on disk is empty... seems the generation is triggered by samba. 
KODI log doesn't give any relevant info, and neither does crashlog. Even tried debug logging.
I was thinking to downgrade the machine, but that doesn't trigger the bug.

Any suggestions? 

I'll see if I can bridge a laptop's WiFi to LAN and connect via cable, and see if samba shares work...

**Comment by dlabz** [comment]
Ok, LAN worked:

```
03:00.0 Network controller [0280]: Broadcom Corporation BCM43225 802.11b/g/n [14e4:4357] (rev 01)
	Subsystem: Foxconn International, Inc. BCM43225 802.11b/g/n (T77H103.00 Wireless Half-size Mini PCIe Card) [105b:e021]
	Control: I/O- Mem+ BusMaster+ SpecCycle- MemWINV- VGASnoop- ParErr- Stepping- SERR- FastB2B- DisINTx-
	Status: Cap+ 66MHz- UDF- FastB2B- ParErr- DEVSEL=fast >TAbort- <TAbort- <MAbort- >SERR- <PERR- INTx-
	Latency: 0, Cache Line Size: 64 bytes
	Interrupt: pin A routed to IRQ 16
	Region 0: Memory at fe000000 (64-bit, non-prefetchable) [size=16K]
	Capabilities: [40] Power Management version 3
		Flags: PMEClk- DSI- D1+ D2+ AuxCurrent=0mA PME(D0+,D1-,D2-,D3hot+,D3cold+)
		Status: D0 NoSoftRst+ PME-Enable- DSel=0 DScale=2 PME-
	Capabilities: [58] Vendor Specific Information: Len=78 <?>
	Capabilities: [48] MSI: Enable- Count=1/1 Maskable- 64bit+
		Address: 0000000000000000  Data: 0000
	Capabilities: [d0] Express (v1) Endpoint, MSI 00
		DevCap:	MaxPayload 128 bytes, PhantFunc 0, Latency L0s <4us, L1 unlimited
			ExtTag+ AttnBtn- AttnInd- PwrInd- RBE+ FLReset- SlotPowerLimit 0.000W
		DevCtl:	Report errors: Correctable- Non-Fatal- Fatal- Unsupported-
			RlxdOrd- ExtTag- PhantFunc- AuxPwr- NoSnoop-
			MaxPayload 128 bytes, MaxReadReq 128 bytes
		DevSta:	CorrErr- UncorrErr- FatalErr- UnsuppReq- AuxPwr+ TransPend-
		LnkCap:	Port #0, Speed 2.5GT/s, Width x1, ASPM L0s L1, Exit Latency L0s <4us, L1 <64us
			ClockPM+ Surprise- LLActRep+ BwNot- ASPMOptComp-
		LnkCtl:	ASPM Disabled; RCB 64 bytes Disabled- CommClk+
			ExtSynch- ClockPM- AutWidDis- BWInt- AutBWInt-
		LnkSta:	Speed 2.5GT/s, Width x1, TrErr- Train- SlotClk+ DLActive+ BWMgmt- ABWMgmt-
	Capabilities: [100 v1] Advanced Error Reporting
		UESta:	DLP- SDES- TLP- FCP- CmpltTO- CmpltAbrt- UnxCmplt- RxOF- MalfTLP- ECRC- UnsupReq- ACSViol-
		UEMsk:	DLP- SDES- TLP- FCP- CmpltTO- CmpltAbrt- UnxCmplt- RxOF- MalfTLP- ECRC- UnsupReq- ACSViol-
		UESvrt:	DLP+ SDES- TLP- FCP+ CmpltTO- CmpltAbrt- UnxCmplt- RxOF+ MalfTLP+ ECRC- UnsupReq- ACSViol-
		CESta:	RxErr- BadTLP- BadDLLP- Rollover- Timeout- NonFatalErr+
		CEMsk:	RxErr- BadTLP- BadDLLP- Rollover- Timeout- NonFatalErr+
		AERCap:	First Error Pointer: 14, GenCap+ CGenEn- ChkCap+ ChkEn-
	Capabilities: [13c v1] Virtual Channel
		Caps:	LPEVC=0 RefClk=100ns PATEntryBits=1
		Arb:	Fixed- WRR32- WRR64- WRR128-
		Ctrl:	ArbSelect=Fixed
		Status:	InProgress-
		VC0:	Caps:	PATOffset=00 MaxTimeSlots=1 RejSnoopTrans-
			Arb:	Fixed- WRR32- WRR64- WRR128- TWRR128- WRR256-
			Ctrl:	Enable+ ID=0 ArbSelect=Fixed TC/VC=01
			Status:	NegoPending- InProgress-
	Capabilities: [160 v1] Device Serial Number 00-00-00-ff-ff-c0-78-e4
	Capabilities: [16c v1] Power Budgeting <?>
	Kernel driver in use: wl
	Kernel modules: wl, bcma
```
So, wl, I'd say. Let me know if you need any more info... I can attach the full log, if it's relevant

**Comment by dlabz** [comment]
P.S.
```
[    7.252727] wl: module license 'MIXED/Proprietary' taints kernel.
[    7.252740] Disabling lock debugging due to kernel taint
[    7.343399] wlan0: Broadcom BCM4357 802.11 Hybrid Wireless Controller 6.30.223.271 (r587334)
```

Found this in the system.log

**Comment by chewitt** [comment]
it looks like two drivers trying to use the same device, so blacklist individual drivers and see which one works best (or works at all), e.g. 

```echo "blacklist bcma" > /storage/.config/modprobe.d/blacklist.conf```

or

```echo "blacklist wl" > /storage/.config/modprobe.d/blacklist.conf```

**Comment by dlabz** [comment]
Thanks, I'll give it a try. 

For now, I've downgraded to OpenELEC 6.0.0, as this box is driving my living room screen.
I can upgrade again and test a bit, but getting a WiFi adapter that that works is a more resource-friendly option for me, considering my schedule.

**Comment by Megachip** [comment]
sounds like the same issue as https://github.com/OpenELEC/OpenELEC.tv/issues/4295
